server/src/test/java/org/bonitasoft/web/rest/server/api/organization/APIUserTest.java

```diff
@@ -37,17 +44,80 @@ public void before() throws Exception {
         ItemDefinitionFactory.setDefaultFactory(mock(ItemDefinitionFactory.class));
         apiUser = spy(new APIUser());
         doReturn(userDatastore).when(apiUser).getUserDatastore();
+        doReturn(TestValidator.class.getName()).when(apiUser).getValidatorClassName();
+        I18n.getInstance();
+        APIServletCall caller = mock(APIServletCall.class);
+        apiUser.setCaller(caller);
+        doReturn("en_US").when(caller).getLocale();
     }
 
     @Test
     public void should_user_be_updated_with_icon_as_submit_by_the_API() throws Exception {
         //given
         doReturn(userItem).when(userDatastore).get(USER_ID);
         //when
-        HashMap<String, String> item = new HashMap<>();
-        item.put(UserItem.ATTRIBUTE_ICON, "theAvatar.jpg");
-        apiUser.update(USER_ID, item);
+        apiUser.update(USER_ID, map(UserItem.ATTRIBUTE_ICON, "theAvatar.jpg"));
         //then
-        verify(userDatastore).update(eq(APIID.makeAPIID(123L)), eq(item));
+        verify(userDatastore).update(eq(USER_ID), eq(map(UserItem.ATTRIBUTE_ICON, "theAvatar.jpg")));
+    }
+
+    private HashMap<String, String> map(String key, String value) {
+        HashMap<String, String> item = new HashMap<>();
+        item.put(key, value);
+        return item;
+    }
+
+    @Test
+    public void should_not_update_password_if_empty() throws Exception {
+        apiUser.update(USER_ID, map(UserItem.ATTRIBUTE_PASSWORD, ""));
+
+        verify(userDatastore).update(eq(USER_ID), eq(Collections.<String, String> emptyMap()));
+    }
+
+    @Test
+    public void should_check_password_robustness_on_update() throws Exception {
+        expectedException.expect(ValidationException.class);
+        //see TestValidator class
+        expectedException.expectMessage("there is an error");
+
+        apiUser.update(USER_ID, map(UserItem.ATTRIBUTE_PASSWORD, "a"));
+    }
+
+    @Test
+    public void should_update_password_if_valid() throws Exception {
+        apiUser.update(USER_ID, map(UserItem.ATTRIBUTE_PASSWORD, "b"));
+
+        verify(userDatastore).update(eq(USER_ID), eq(map(UserItem.ATTRIBUTE_PASSWORD, "b")));
+    }
+
+    @Test
+    public void should_check_password_robustness_on_add() throws Exception {
+        UserItem userItem = new UserItem();
+        userItem.setUserName("John");
+        userItem.setPassword("a");
+
+        expectedException.expect(ValidationException.class);
+        //see TestValidator class
+        expectedException.expectMessage("there is an error");
+        apiUser.add(userItem);
     }
+
+    @Test
+    public void should_throw_exception_when_adding_a_user_with_no_password() throws Exception {
+        UserItem userItem = new UserItem();
+        userItem.setUserName("John");
+
+        expectedException.expect(ValidationException.class);
```

Comment:
yep, in that case it would be ok. But still, since we use the expectedException rule to check the message in other tests I think it is better to use the it.
But that is more a philosophical issue.